awx/main/dispatch/control.py

```diff
@@ -20,40 +20,40 @@ def __init__(self, service, host=None):
             raise RuntimeError('{} must be in {}'.format(service, self.services))
         self.service = service
         self.queuename = host or get_local_queuename()
-        self.queue = Queue(self.queuename, Exchange(self.queuename), routing_key=self.queuename)
 
     def publish(self, msg, conn, **kwargs):
-        producer = Producer(
-            exchange=self.queue.exchange,
-            channel=conn,
-            routing_key=self.queuename
-        )
-        producer.publish(msg, expiration=5, **kwargs)
+        # TODO: delete this method??
+        raise RuntimeError("Publish called?!")
 
     def status(self, *args, **kwargs):
         return self.control_with_reply('status', *args, **kwargs)
 
     def running(self, *args, **kwargs):
         return self.control_with_reply('running', *args, **kwargs)
 
+    @classmethod
+    def generate_reply_queue_name(cls):
+        letters = string.ascii_lowercase
+        return 'reply_to_{}'.format(''.join(random.choice(letters) for i in range(8)))
```

Comment:
Is there a reason for `range(8)`?  This seems _potentially_ susceptible to collisions?  Would a `uuid4()` be better, or is this a limitation of pg's pub/sub?